src/types/Theme.ts

```diff
@@ -23,6 +23,20 @@ interface TextShadowParams {
   textShadowRadius: number
 }
 
+interface ButtomParamsUi4 {
+  textStyle: {
+    fontFamily: string
+    color: string
+  }
+  gradientProps: LinearGradientProps
+  shadowParams: ThemeShadowParams // TODO: Address android/ios inconsistencies with lib
```

Comment:
We need to specify exactly what's allowed in the gradient options, so please define `ThemeGradientProps` like you have defined `ThemeShadowParams` or even the loose `containerStyle` below. That way we only have colors in the theme, not arbitrary code.

Reply:
Actually on second thought - for this one, I intentionally specified
LinearGradientProps because I wanted the flexibility of controlling any aspect
of the gradient, not just a subset. In the implementation, I splat out the theme props.